.github/workflows/check-coverage.yml

```diff
@@ -21,6 +32,14 @@ jobs:
       - name: Remove mocks.go lines
         run: |
           sed '/mocks.go/d' coverage_raw.out > coverage.out
+      - name: Az CLI login
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.GO_E2E_CLIENT_ID }}
+          tenant-id: ${{ secrets.GO_E2E_TENANT_ID }}
+          subscription-id: ${{ secrets.GO_E2E_SUBSCRIPTION_ID }}
+      - name: Run e2e tests
+        run: cd e2e; go test -timeout 30m -v -run Test_All ./...
```

Comment:
is there a way we can reference e2e.yaml which is already used by the ADO pipeline to run the tests? This would be the 3rd instance of this run command spread across different files (already one in e2e.yaml and one in e2e-local-go.sh)

Reply:
hmmmm, I'm also thinking it could be useful to implement a 'dry-run' mode for AB E2E that wouldn't actually interact with Azure very much (if at all) so we could keep these coverage checks reasonably light-weight and localized - would also make sense since really the coverage that comes from these E2E tests is only directly dependent on the NodeBootstrappingConfiguration which gets fed into the API (dry-run could basically just feed the nbc into agentbaker and verify no panics/errors occur, similar to fuzz)